src/core/a-entity.js

```diff
@@ -423,15 +425,15 @@ var proto = Object.create(ANode.prototype, {
   },
 
   removeComponent: {
-    value: function (name) {
+    value: function (name, force) {
```

Comment:
An `a-scene` without a `canvas` component does not make sense either. That's why is a default component and we don't allow removal. I would have a `restore` component that can save the original state of all the components you care about and restore the values via `setAttribute` instead of removing / adding components. The logic is not very complicated and it will yield better performance. The default components are there for a reason and a mechanism to workaround the mechanism introduces a lot of inconsistencies and potential for misuse.